Fix TAPI verification errors on ARM64_32 builds.

MenloDorian opened the pull request:
#### 501c1d5fe26dba28fcd1b4584a367caaebc909a0
<pre>
Fix TAPI verification errors on ARM64_32 builds.
<a href="https://bugs.webkit.org/show_bug.cgi?id=301453">https://bugs.webkit.org/show_bug.cgi?id=301453</a>
<a href="https://rdar.apple.com/163372818">rdar://163372818</a>

Reviewed by Keith Miller and Yusuke Suzuki.

The fix entails:
1. Change pas_config.h&apos;s default setting of LIBPAS_ENABLED to 1 only apply when PAS_CPU(ADDRESS64).
   Otherwise default it to 0.
2. Add #if LIBPAS_ENABLED + #endif around the content of many libpas header files that TAPI
   complains about for ARM64_32 builds.

libpas does not support ARM64_32 targets.  Hence, on ARM64_32 builds, libpas is disabled.  So, it
is correct to #if out all the libpas code in those header files.

No new tests needed because the build verifies the fix.
.
* Source/bmalloc/libpas/src/libpas/bmalloc_heap_config.h:
* Source/bmalloc/libpas/src/libpas/bmalloc_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/hotbit_heap_config.h:
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_tagged_page_granule_use_count_ptr.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_tagged_unsigned_ptr.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_thread_local_cache_layout_node.h:
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_deallocate.h:
* Source/bmalloc/libpas/src/libpas/pas_designated_intrinsic_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_exclusive_view_template_memo_table.h:
* Source/bmalloc/libpas/src/libpas/pas_full_alloc_bits_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout_entry.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_array.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_common.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_intrinsic.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_primitive.h:
* Source/bmalloc/libpas/src/libpas/pas_try_reallocate.h:

Canonical link: <a href="https://commits.webkit.org/302174@main">https://commits.webkit.org/302174@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2b842f5341a5972a28898ff8ab3b93af9336fb3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128094 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/377 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38926 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/135461 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79588 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c2c2de58-aa91-4bdb-b572-1584329093fa) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/295 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/250 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97516 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/65398 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1f176060-17b2-451b-9252-d05ca6e0e018) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131042 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/174 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114751 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78085 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/86aec8b6-d003-4461-a9f9-1b2baed22757) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/166 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/32857 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/78771 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/120128 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108541 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/33344 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137951 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/126556 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/230 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/222 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106044 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/261 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111094 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105782 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/172 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29648 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/52410 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20036 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/277 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/159576 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/186 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39832 "Found 16 new JSC stress test failures: wasm.yaml/wasm/stress/osr-entry-live-fpr.js.default-wasm, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-bbq, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-bbq-no-consts, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-collect-continuously, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-eager, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-eager-jettison, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-no-cjit, wasm.yaml/wasm/stress/osr-entry-live-fpr.js.wasm-slow-memory, wasm.yaml/wasm/stress/osr-entry-live-stack.js.default-wasm, wasm.yaml/wasm/stress/osr-entry-live-stack.js.wasm-bbq ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/260 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/236 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/bmalloc/libpas/src/libpas/bmalloc_heap_config.h:
```diff
@@ -87,5 +88,5 @@ PAS_END_EXTERN_C;
 
+#endif /* LIBPAS_ENABLED */
 #endif /* BMALLOC_HEAP_CONFIG_H */
-
```
Is there a reason to remove the newline at the end of these files? I thought that sometimes ended up being important if e.g. some file starts with a non-comment.

MenloDorian replied:
I saw 2 newlines.  So, I removed 1.  Did I removed all of it?  I can check again

MenloDorian replied:
I've confirmed it.  All these files are still terminated with a newline.  Previously, they had 2 newlines.  So, all is well.

kmiller68 replied:
Ah, ok, sorry for the noise. I guess that explains why it didn't have the flag...

kmiller68 reviewed (approved):
r=me with question

Commit:
Fix TAPI verification errors on ARM64_32 builds.
https://bugs.webkit.org/show_bug.cgi?id=301453
rdar://163372818

Reviewed by Keith Miller and Yusuke Suzuki.

The fix entails:
1. Change pas_config.h's default setting of LIBPAS_ENABLED to 1 only apply when PAS_CPU(ADDRESS64).
   Otherwise default it to 0.
2. Add #if LIBPAS_ENABLED + #endif around the content of many libpas header files that TAPI
   complains about for ARM64_32 builds.

libpas does not support ARM64_32 targets.  Hence, on ARM64_32 builds, libpas is disabled.  So, it
is correct to #if out all the libpas code in those header files.

No new tests needed because the build verifies the fix.
.
* Source/bmalloc/libpas/src/libpas/bmalloc_heap_config.h:
* Source/bmalloc/libpas/src/libpas/bmalloc_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/hotbit_heap_config.h:
* Source/bmalloc/libpas/src/libpas/pas_bitfit_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_tagged_page_granule_use_count_ptr.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_tagged_unsigned_ptr.h:
* Source/bmalloc/libpas/src/libpas/pas_compact_thread_local_cache_layout_node.h:
* Source/bmalloc/libpas/src/libpas/pas_config.h:
* Source/bmalloc/libpas/src/libpas/pas_deallocate.h:
* Source/bmalloc/libpas/src/libpas/pas_designated_intrinsic_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_exclusive_view_template_memo_table.h:
* Source/bmalloc/libpas/src/libpas/pas_full_alloc_bits_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_heap_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_exclusive_view_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_size_directory_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_allocator_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout.h:
* Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout_entry.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_array.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_common.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_intrinsic.h:
* Source/bmalloc/libpas/src/libpas/pas_try_allocate_primitive.h:
* Source/bmalloc/libpas/src/libpas/pas_try_reallocate.h:

Canonical link: https://commits.webkit.org/302174@main
